### PR TITLE
refactor: Update variables and outputs to align with `terraform-aws-eks` module

### LINKED ISSUE #779: Align variables and outputs 1:1 with `terraform-aws-eks` module
#### Is your feature request related to a problem? Please describe
- Align the variables and outputs between the EKS Blueprints project and the `terraform-aws-eks` module

#### Describe the solution you'd like
- Since EKS Blueprints wraps the `terraform-aws-eks` module, to make it more of a consistent user experience, we should use align the variable names and outputs. 
	- For example, the EKS Blueprint output [`worker_node_security_group_arn`](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/9b9686dcb8cc7a2a3717b4e582de39e6d60fabdf/outputs.tf#L75) points to the `terraform-aws-eks` module's [`node_security_group_arn`](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/9b9686dcb8cc7a2a3717b4e582de39e6d60fabdf/outputs.tf#L77) output. The EKS Blueprint output should be updated to match (`node_security_group_arn`)

#### Additional context
- Align variables and outputs 1:1 with `terraform-aws-eks` module

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?
- Update variables and outputs to align with `terraform-aws-eks` module
- Remove `{ "kubernetes.io/cluster/${local.name}" = "shared" }` tags from VPCs; these are no longer required as of EKS 1.19 and v2.1.1 > of AWS LBC https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html#network-requirements-subnets
- `CODEOWNERS` file moved under `.github` directory; works the same, just keeps project root cleaner/slimmer
- `OWNERS` file is removed - found out this is a K8s prow artifact and not used in AWS
- tflint version updated to latest to catch additional lint checks

### Motivation

- Closes #779 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,60 +22,6 @@ For complete project documentation, please visit our [documentation site](https:
 
 To view examples for how you can leverage EKS Blueprints, please see the [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) directory.
 
-## Usage
```
Comment (bryantbiggs) [inline README.md:25]: Not necessary to keep up to date since we have all the examples

File: examples/ai-ml/ray/main.tf [hunk 2]
```diff
@@ -3,30 +3,16 @@ provider "aws" {
 }
 
 provider "kubernetes" {
-  host                   = module.eks_blueprints.eks_cluster_endpoint
-  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+  host                   = module.eks_blueprints.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.cluster_certificate_authority_data)
   token                  = data.aws_eks_cluster_auth.this.token
-
-  exec {
```
Comment (bryantbiggs) [inline examples/ai-ml/ray/main.tf:10]: aligns provider auth with rest of examples (auth token route)

File: examples/ai-ml/ray/main.tf [hunk 3]
```diff
@@ -3,30 +3,16 @@ provider "aws" {
...
 provider "helm" {
   kubernetes {
-    host                   = module.eks_blueprints.eks_cluster_endpoint
-    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+    host                   = module.eks_blueprints.cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks_blueprints.cluster_certificate_authority_data)
     token                  = data.aws_eks_cluster_auth.this.token
-
-    exec {
-      api_version = "client.authentication.k8s.io/v1beta1"
```
Comment (bryantbiggs) [inline examples/ai-ml/ray/main.tf:25]: aligns provider auth with rest of examples (auth token route)

File: examples/ai-ml/ray/main.tf [hunk 4]
```diff
@@ -210,7 +197,7 @@ module "s3_bucket" {
 data "aws_iam_policy_document" "irsa_policy" {
   statement {
     actions   = ["s3:ListBucket"]
-    resources = ["${module.s3_bucket.s3_bucket_arn}"]
```
Comment (bryantbiggs) [inline examples/ai-ml/ray/main.tf:213]: new tflint version caught this

File: examples/karpenter/main.tf [hunk 5]
```diff
@@ -175,7 +175,7 @@ module "eks_blueprints_kubernetes_addons" {
 }
 
 resource "kubectl_manifest" "karpenter_provisioners" {
-  yaml_body = templatefile("provisioners/karpenter_provisioners.yaml", {
+  yaml_body = templatefile("manifests/karpenter_provisioners.yaml", {
```
Comment (bryantbiggs) [inline examples/karpenter/main.tf:178]: Fixes Karpenter example that got mixed up somewhere along in the refactor

File: modules/irsa/main.tf [hunk 6]
```diff
@@ -32,7 +32,7 @@ resource "aws_iam_role" "irsa" {
       {
         "Effect" : "Allow",
         "Principal" : {
-          "Federated" : "${var.eks_oidc_provider_arn}"
```
Comment (bryantbiggs) [inline modules/irsa/main.tf:35]: new tflint version catch

File: modules/kubernetes-addons/csi-secrets-store-provider-aws/README.md [hunk 7]
```diff
@@ -10,16 +10,20 @@ AWS Secrets Manager and Config Provider for Secret Store CSI Driver allows you t
 
 See the [csi-secrets-store-provider-aws](https://github.com/aws/eks-charts/tree/master/stable/csi-secrets-store-provider-aws).
 
-<!--- BEGIN_TF_DOCS --->
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/csi-secrets-store-provider-aws/README.md:13]: Outdated terraform-docs flag so docs were out of sync (see updates directly below)

File: modules/kubernetes-addons/secrets-store-csi-driver/README.md [hunk 8]
```diff
@@ -14,16 +14,20 @@ For more details, refer [Secrets Store CSI Driver](https://secrets-store-csi-dri
 
 See the [secrets-store-csi-driver](https://secrets-store-csi-driver.sigs.k8s.io/getting-started/installation.html).
 
-<!--- BEGIN_TF_DOCS --->
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/secrets-store-csi-driver/README.md:17]: Another outdated terraform-docs flag - docs now updated

File: outputs.tf [hunk 9]
```diff
@@ -1,141 +1,214 @@
...
+  description = "Map of attribute maps for all self managed node groups created"
+  value       = module.eks.self_managed_node_groups
+}
+
+output "self_managed_node_groups_autoscaling_group_names" {
+  description = "List of the autoscaling group names created by self-managed node groups"
+  value       = module.eks.self_managed_node_groups_autoscaling_group_names
 }
 
 #-------------------------------
```
Comment (bryantbiggs) [inline outputs.tf:190]: Everything above is direct copy+paste from `terraform-aws-eks` module

File: examples/ai-ml/ray/main.tf (hunk 4 above)
Comment (askulkarni2) [reply examples/ai-ml/ray/main.tf:213]: nice

Comment (askulkarni2) [reply examples/ai-ml/ray/main.tf:213]: nice

File: main.tf [hunk 10]
```diff
@@ -94,7 +127,7 @@ module "emr_on_eks" {
   }
 
   emr_on_eks_teams              = each.value
-  eks_cluster_id                = module.aws_eks.cluster_id
+  eks_cluster_id                = module.eks.cluster_id
```
Comment (askulkarni2) [inline main.tf:130]: @vara-bonthu does it make sense to restructure emr_on_eks as an add-on?

File: modules/kubernetes-addons/main.tf [hunk 11]
```diff
@@ -135,7 +135,7 @@ module "aws_load_balancer_controller" {
 }
 
 module "aws_node_termination_handler" {
-  count                   = var.enable_aws_node_termination_handler && (length(var.auto_scaling_group_names) > 0 || var.enable_karpenter) ? 1 : 0
+  count                   = var.enable_aws_node_termination_handler ? 1 : 0
```
Comment (askulkarni2) [inline modules/kubernetes-addons/main.tf:138]: Yikes, that was an odd conditional, glad its being removed.

**Review by askulkarni2 — APPROVED** [review]
Massive. LGTM! Likely needs a full example CI run.

**Pull request merged**